### PR TITLE
Ignore line anchor links with leading zeroes

### LINKED ISSUE #21722: Jump to Line Fragment Should Ignore Line Numbers < 1
### Description

When calling https://codeberg.org/Codeberg/gitea/src/branch/codeberg-1.17/README.md?display=source#L0  it is internally redirected to https://codeberg.org/Codeberg/gitea/src/branch/codeberg-1.17/README.md?display=source#undefined
which leads to the following javascript error:

>JavaScript error: Uncaught TypeError: Cannot read properties of undefined (reading 'length') (https://codeberg.org/assets/js/index.js?v=1.17.3~87-ge4b8341 @ 83:11976). Open browser console to see more details.

Maybe line numbers < 1 can simply be ignored as github does:

https://github.com/go-gitea/gitea/blob/main/README.md?plain=1#L0

This would fix issues with links from external systems that don't check the line number properly.

### Gitea Version

1.17.3

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

helm chart

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by silverwind**
Fixes: https://github.com/go-gitea/gitea/issues/21722

### CODE REVIEW

File: web_src/js/features/repo-code.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ import {createTippy, showTemporaryTooltip} from '../modules/tippy.js';
 import {copyToClipboard} from './clipboard.js';
 
 const {i18n} = window.config;
+export const singleRe = /^#(L|n)([1-9][0-9]*)$/;
+export const rangeRe = /^#(L[1-9][0-9]*)-(L[1-9][0-9]*)$/;
```
Comment (delvh) [inline web_src/js/features/repo-code.js]: What about
```suggestion
export const singleLineAnchorRegex = /^#(L|n)([1-9][0-9]*)$/;
export const lineRangeAnchorRegex = /^#(L[1-9][0-9]*)-(L[1-9][0-9]*)$/;
```
?

**Review by lunny — APPROVED** [review]

**Review by delvh — APPROVED** [review]

File: web_src/js/features/repo-code.js (hunk 1 above)
Comment (silverwind) [reply web_src/js/features/repo-code.js]: Seems better to rename, but I think we have the `Re` suffix in use in other places, but will check.

Comment (silverwind) [reply web_src/js/features/repo-code.js]: Renamed to `singleAnchorRegex` and `rangeAnchorRegex`.

```
$ rg 'Regex =' | wc -l
15
$ rg 'Re =' | wc -l
4

**Comment by lunny** [comment]
make L-G-T-M work

**Pull request merged**